lib/grant-types/refresh-token-grant-type.ts

```diff
@@ -0,0 +1,158 @@
+import { AbstractGrantType } from '.';
+import {
+  InvalidArgumentError,
+  InvalidGrantError,
+  InvalidRequestError,
+  ServerError,
+} from '../errors';
+import { Client, RefreshToken, User } from '../interfaces';
+import { Request } from '../request';
+import * as is from '../validator/is';
+
+export class RefreshTokenGrantType extends AbstractGrantType {
+  constructor(options: any = {}) {
+    super(options);
+
+    if (!options.model) {
+      throw new InvalidArgumentError('Missing parameter: `model`');
+    }
+
+    if (!options.model.getRefreshToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `getRefreshToken()`',
+      );
+    }
+
+    if (!options.model.revokeToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `revokeToken()`',
+      );
+    }
+
+    if (!options.model.saveToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `saveToken()`',
+      );
+    }
+  }
```

Comment:
These various stanzas seem somewhat repetitive and almost feel like they could be handled more declaratively (in the abstract grant type assuming that can implement methods). It fells DRY-ish.

Reply:
This is a security relevant module. As such it should be more expressive than anything else.

Reply:
Even though, i would like to have a typeof check too

e.g. 

typeof options.model.saveToken === "function" so that we actually ensure, that the methods are actually methods and not basic datatypes.